vm/evm.go

```diff
@@ -260,6 +260,10 @@ func (evm *EVM) Call(ctx context.Context, caller ethvm.ContractRef, addr common.
 			Block: &types.BlockInput{Number: &blockNum},
 		})
 		if preCallResult.Err != nil {
+			if preCallResult.Err.Error() == ErrOutOfGas.Error() {
```

Comment:
Why do we need to reset the error here? It doesn't seem like there's much change in definition of preCallResult.